.rhcicd/clowdapp-backend.yaml

```diff
@@ -183,25 +183,91 @@ objects:
       secretName: ${FLOORIST_BUCKET_SECRET_NAME}
     suspend: ${{FLOORIST_SUSPEND}}
     queries:
-    - prefix: insights/notifications/splunk
+    # Count the number of created endpoints per endpoint type. Raw total, which
+    # includes the disabled ones or the ones not assigned in a behavior group.
+    - prefix: insights/notifications/endpoint_types
       query: >-
         SELECT
-          DISTINCT "account_id"
-          FROM "endpoints"
-          WHERE "endpoint_type_v2"='CAMEL' AND "endpoint_sub_type"='splunk';
-    - prefix: insights/notifications/accounts
+          CASE
+            WHEN
+              endpoint_type_v2 = 'CAMEL'
+            THEN
+              LOWER(endpoint_sub_type)
+            ELSE
+              LOWER(endpoint_type_v2)
+          END AS "Endpoint type",
+          count(endpoint_type_v2) AS "Count"
+        FROM
+          endpoints
+        GROUP BY
+          endpoint_type_v2,
+          endpoint_sub_type;
+    # Count the number of enabled endpoints per endpoint type. Same as above,
+    # but in this case just counting the enabled ones.
+    - prefix: insights/notifications/endpoint_types
       query: >-
         SELECT
-        "applications"."id", "applications"."name", COUNT("applications"."id") AS "applications_count"
-        FROM "applications"
-        INNER JOIN "endpoint_email_subscriptions" ON "applications"."id"="endpoint_email_subscriptions"."application_id"
-        GROUP BY "applications"."id" HAVING COUNT("applications"."id") > 1;
-    - prefix: insights/notifications/accounts
+          CASE
+            WHEN
+              endpoint_type_v2 = 'CAMEL'
+            THEN
+              LOWER(endpoint_sub_type)
+            ELSE
+              LOWER(endpoint_type_v2)
+          END AS "Enabled endpoint type",
+          count(endpoint_type_v2) AS "Count"
+        FROM
+          endpoints
+        WHERE
+          enabled=true
+        GROUP BY endpoint_type_v2;
+    # Count the number of enabled endpoints per type which are assigned to a
+    # behavior group, and therefore considered to be in active use.
+    - prefix: insights/notifications/endpoint_types
       query: >-
         SELECT
-        COUNT(distinct account_id)
-        FROM endpoints
-        WHERE endpoint_type_v2 = 'CAMEL' AND endpoint_sub_type = 'splunk' AND enabled IS FALSE;
+          CASE
+            WHEN
+              e.endpoint_type_v2 = 'CAMEL'
+            THEN
+              LOWER(e.endpoint_sub_type)
+            ELSE
+              LOWER(e.endpoint_type_v2)
+          END AS "Endpoint type",
+          count(e.endpoint_type_v2) AS "Count"
+        FROM
+          endpoints AS e
+        INNER JOIN
+          behavior_group_action AS bga ON bga.endpoint_id = e.id
+        INNER JOIN
+            event_type_behavior AS etb ON etb.behavior_group_id = bga.behavior_group_id
```

Comment:
Instead of running 3 queries for endpoints, we could run a single one that would return all the information:
| Endpoint type | Count | Enabled | Actively used |
| - | - | - | - |
| Slack | 18 | true | true |
| Slack | 7 | true | false |
[...]

That kind of data structure can easily by aggregated and/or filtered on any column using a pivot table, which is one of the basic tools used in business intelligence.

Reply:
I have come up with the following query, but grouping by the endpoint's ID —which is required to be used in the subquery, since otherwise SQL complains— makes lots of duplicated results —obviously—. I am trying to figure out a better way, but I'll leave the query here just in case you spot something:

```sql
SELECT
  -- When the endpoint is a Camel endpoint, get its subtype instead.
  CASE
    WHEN
      e.endpoint_type_v2 = 'CAMEL'
    THEN
      LOWER(e.endpoint_sub_type)
    ELSE
      LOWER(e.endpoint_type_v2)
  END AS "Endpoint type",
  COUNT(*) AS "Count",
  e.enabled AS "Enabled",
  -- If the endpoint has an associated event type and a behavior group, then
  -- that means that it is actively being used.
  EXISTS (
    SELECT
        1
    FROM
        behavior_group_action AS bga
    INNER JOIN
        event_type_behavior AS etb ON etb.behavior_group_id = bga.behavior_group_id
    WHERE
        bga.endpoint_id = e.id 
) AS "Actively used"
FROM
  endpoints AS e
WHERE
  e.enabled=TRUE
GROUP BY
  e.endpoint_type_v2,
  e.endpoint_sub_type,
  e.enabled,
  e.id
```

Reply:
I have left the results of this query after running it in stage in the ticket —restricted to Red Hat employee for security purposes—.